source/blog/2017-02-06-finagle-loves-netty4.md

```diff
@@ -0,0 +1,76 @@
+---
+layout: post
+title: Netty 4 support in Finagle
+published: true
+post_author:
+  display_name: Vladimir Kostyukov
+  twitter: vkostyukov
+tags: Finagle, Netty4
+---
+
+It's been quite a while since the [Netty 4 migration in Finagle][netty4-blog-post] was initially
+announced. We've travelled a long way and are happy to announce that there is now (as per
+[Finagle 6.42][6.42]) support for Netty 4 transports in most of the protocols: Thrift, ThriftMux,
+Memcached, MySQL, Kestrel, and Redis.
+
+## How to enable Netty 4?
+
+While we have not yet defaulted to Netty 4, we've been running it in production for several months
+and have gained enough confidence to publicize the availability of the alternative transports in
+Finagle.
+
+We encourage Finagle users to try out the new Netty 4 transports for their protocols and jump on the
+fast track to upcoming changes around resiliency (think of HTTP/2) and performance (think of reduced
+allocation profile and better threading model in Netty 4).
+
+To switch the transport over to Netty 4 supply the following command line flag:.
+
+```
+-Dcom.twitter.finagle.toggle.flag.overrides=com.twitter.$protocol.UseNetty4=1.0
+```
+
+Where `$protocol` is one of the following: `mux` (use this for ThriftMux), `thrift`, `mysql`,
+`memcached`, `kestrel`.
+
+This command line flag overrides [a feature toggle][toggles] that is evaluated at application
+start up and is *global for all clients/servers running on the same JVM instance*.
+
+Note that Netty 4 is already enabled by default in `finagle-redis` so no need for an extra CLI flag.
+
+## What about HTTP/1.1?
+
+HTTP/1.1 on Netty 4 is still a work in progress. There are known limitations for HTTP clients, but
+we've been successfully running `finagle-http` servers with Netty 4 in production for several weeks.
```

Comment:
we should note that TwitterServer's admin http server has been running it for months.